lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3525,4 +3525,17 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Determines if a string literal is valid as an identifier in this
+     * platform.
+     *
+     * @param string $name The string literal to test
+     *
+     * @return bool
+     */
+    public function isValidIdentifier($name)
```

Comment:
@zeroedin-bill the issue is totally valid and I also agree that identifiers should be quoted intelligently by the schema manager when building the schema objects. Just not sure what the proper approach is here. Maybe we should leave that functionality completely internal.
I think I have done something similar for Oracle already. See [here](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Schema/OracleSchemaManager.php#L339-L356). Although this was more about casing, it is somewhat related as it "normalizes" identifiers coming from the DB to be usable by Doctrine and to correlate better with mappings and such...